source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/Util.java

```diff
@@ -35,13 +35,30 @@
 
 public final class Util {
 
+    /**
+     * Convert dp to px
+     * @param context
+     * @param dp The number of Android dips (display-independent pixels)
+     * @return The number of equivalent physical pixels
+     */
     public static int dpToPixels(Context context, long dp)
     {
         DisplayMetrics metrics = context.getResources().getDisplayMetrics();
         int returnVal = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, metrics);
         return returnVal;
     }
 
+    /**
+     * Convert px to dp
+     * @param context
+     * @param pixels The number of physical pixels
+     * @return The number of equivalent Android dips (display-independent pixels)
+     */
+    public static double pixelsToDp(Context context, double pixels)
+    {
+        return pixels / dpToPixels(context, 1);
+    }
```

Comment:
If the divisor is 0 (this only happens if the screen density is absurdly low like 8 pixels per inch), what would be a good value to return here? Probably best to allow it to become Infinity, because we don't have a better value to give at that point.

Reply:
Didn't realize Java doesn't throw when divided by 0.0. dpToPixels returns an integer, so anything less than 1 > and  > 0, will become infinity, but if the value is 2 > and 1 >= becomes pixels. so 1 seems to be more a reasonable value.

Reply:
Per the Java spec, floating-point division by zero resolves to positive or negative infinity:

`for example, 1.0/0.0 has the value positive infinity, while the value of 1.0/-0.0 is negative infinity`

https://docs.oracle.com/javase/specs/jls/se8/html/jls-4.html#jls-4.2.3

Reply:
thanks